base/applications/mspaint/atlimagedx.h

```diff
@@ -7,94 +7,72 @@
 
 #pragma once
 
+#include <atlimage.h>
+
 class CImageDx : public CImage
 {
 public:
-    CImageDx() : CImage()
-    {
-        GetImageHorizontalResolution = NULL;
-        GetImageVerticalResolution = NULL;
-        BitmapSetResolution = NULL;
-    }
+    using CImage::FileTypeFromExtension;
+    using CImage::CImage;
 
-    BOOL GetResolution(Gdiplus::GpImage *pImage, float *pxDpi, float *pyDpi)
+    HRESULT LoadDx(LPCTSTR pszFileName, float *pxDpi, float *pyDpi)
     {
-        *pxDpi = 96;
-        *pyDpi = 96;
-
-        if (GetImageHorizontalResolution == NULL || GetImageVerticalResolution == NULL)
-        {
-            GetImageHorizontalResolution =
-                AddrOf<GETIMAGEHORIZONTALRESOLUTION>("GdipGetImageHorizontalResolution");
-            GetImageVerticalResolution =
-                AddrOf<GETIMAGEVERTICALRESOLUTION>("GdipGetImageVerticalResolution");
-        }
-
-        if (GetImageHorizontalResolution == NULL || GetImageVerticalResolution == NULL)
-            return FALSE;
-
-        GetImageHorizontalResolution(pImage, pxDpi);
-        GetImageVerticalResolution(pImage, pyDpi);
-        return TRUE;
-    }
-
-    BOOL SetResolution(Gdiplus::GpBitmap *pBitmap, float xDpi, float yDpi) const
-    {
-        if (BitmapSetResolution == NULL)
-            BitmapSetResolution = AddrOf<BITMAPSETRESOLUTION>("GdipBitmapSetResolution");
-
-        if (BitmapSetResolution == NULL)
-            return FALSE;
-
-        BitmapSetResolution(pBitmap, xDpi, yDpi);
-        return TRUE;
-    }
+        using namespace Gdiplus;
 
-    HRESULT LoadDx(LPCTSTR pszFileName, float *pxDpi, float *pyDpi) throw()
-    {
         // convert the file name string into Unicode
         CStringW pszNameW(pszFileName);
 
-        // create a GpBitmap object from file
-        using namespace Gdiplus;
-        GpBitmap *pBitmap = NULL;
-        if (GetCommon().CreateBitmapFromFile(pszNameW, &pBitmap) != Ok)
+        if (pxDpi)
+            *pxDpi = 96;
```

Comment:
Not using the function calls as done in the previous PR you merged?

Reply:
Deleted.